Examples/Physics_applications/laser_ion/README.rst

```diff
@@ -52,14 +50,53 @@ For `MPI-parallel <https://www.mpi-forum.org>`__ runs, prefix these lines with `
 Analyze
 -------
 
-.. note::
+.. _fig-tnsa-ps-electrons-pinhole:
+
+.. figure:: https://user-images.githubusercontent.com/5416860/295003882-c755fd47-4bb3-4439-9319-c48214cbaafd.png
+   :alt: Longitudinal phase space of forward-flying electrons in a 2 degree opening angle.
+   :width: 100%
+
```

Comment:
The `:alt:` parameter does not specify the caption text, and only appears if the image doesn't load. https://sublime-and-sphinx-guide.readthedocs.io/en/latest/images.html#alt-text

Therefore, you need to repeat the text without the `:alt:` parameter in order to actually have a caption. Furthermore, make sure that there is an empty new line above the caption text. In this case, it should look like

```
.. figure:: https://user-images.githubusercontent.com/5416860/295003882-c755fd47-4bb3-4439-9319-c48214cbaafd.png
   :alt: Longitudinal phase space of forward-flying electrons in a 2 degree opening angle.
   :width: 100%

   Longitudinal phase space of forward-flying electrons in a 2 degree opening angle.

.. _fig-tnsa-ps-protons-pinhole:

```

I don't know why the empty line above the caption text is needed, but that's what I found via testing. As a bonus, this also seemed to fix the "undefined label" error coming from `:numref:`.

Reply:
Oh wait! But the CI-built docs now have `:numref:` working!

Thanks, @eebasso!

(Now I only need to figure out why locally that does not fix it 🤔)